python/lsst/ts/standardscripts/auxtel/latiss.py

```diff
@@ -0,0 +1,244 @@
+        """Take a series of bias images.
+
+        Parameters
+        ----------
+        exptime : `float`
+            Exposure time for darks.
+        ndarks : `int`
+            Number of dark frames to take.
+        checkpoint : `coro`
+            A optional coroutine that accepts one string argument that is
+            called before each dark is taken.
+
+        """
+        for i in range(ndarks):
+            tag = f'dark_{i+1:04}_{ndarks:04}'
+            if checkpoint is not None:
+                await checkpoint(tag)
+            await self.expose(exp_time=exptime, shutter=False,
+                              image_seq_name=tag,
+                              science=True, guide=False, wfs=False)
+
+    async def take_flats(self, exptime, nflats,
+                         latiss_filter=None, latiss_grating=None, latiss_linear_stage=None,
+                         checkpoint=None):
+        """Take a series of flat field images.
+
+        Parameters
+        ----------
+        exptime : `float`
+            Exposure time for flats.
+        nflats : `int`
+            Number of flat frames to take.
+        latiss_filter : `None` or `int` or `str`
+            Filter id or name. If None, ignore setting filter.
+        latiss_grating : `None` or `int` or `str`
+            Grating id or name.  If None, ignore setting grating.
+        latiss_linear_stage : `None` or `float`
+            Linear stage position.  If None, ignore setting linear stage.
+        checkpoint : `coro`
+            A optional coroutine that accepts one string argument that is
+            called before each flat field image is taken.
+
+        """
+        for i in range(nflats):
+            tag = f"flat_{i+1:04}"
+            if checkpoint is not None:
+                await checkpoint(tag)
+            await self.take_image(exptime=exptime, shutter=True, image_seq_name=tag,
+                                  latiss_filter=latiss_filter, latiss_grating=latiss_grating,
+                                  latiss_linear_stage=latiss_linear_stage)
+
+    async def take_image(self, exptime, shutter, image_seq_name,
+                         latiss_filter=None, latiss_grating=None, latiss_linear_stage=None,
+                         science=True, guide=False, wfs=False,
+                         ):
+        """Encapsulates the action of setting up the spectrograph and taking
+        images.
+
+        Setting up the spectrograph and taking images cannot be done
+        concurrently. One needs first to setup the spectrograph then,
+        request images.
+
+        Parameters
+        ----------
+        exptime : `float`
+            The exposure time for the image, in seconds.
+        shutter : `bool`
+            Should activate the shutter? (False for bias and dark)
+        image_seq_name : `str`
+            A string to identify the image.
+        latiss_filter : `None` or `int` or `str`
```

Comment:
Why the "latiss_" prefix? That sounds like an implementation detail, and if so, I suggest removing it.

Reply:
The problem is that `filter` is a reserved word in python. 

```
filter?
Init signature: filter(self, /, *args, **kwargs)
Docstring:
filter(function or None, iterable) --> filter object

Return an iterator yielding those items of iterable for which function(item)
is true. If function is None, return the items that are true.
Type:           type
Subclasses:
```

Reply:
How about `filterid`?